src/cascadia/UnitTests_TerminalCore/ScreenSizeLimitsTest.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+#include <WexTestClass.h>
+#include <climits>
+
+#include "../cascadia/TerminalCore/Terminal.hpp"
+#include "../renderer/inc/DummyRenderTarget.hpp"
+#include "consoletaeftemplates.hpp"
+
+#include "winrt/Microsoft.Terminal.Settings.h"
+
+using namespace winrt::Microsoft::Terminal::Settings;
+using namespace Microsoft::Terminal::Core;
+
+namespace TerminalCoreUnitTests
+{
+class MockTermSettings : public winrt::implements<MockTermSettings, ICoreSettings>
+{
+public:
+    MockTermSettings(int32_t historySize, int32_t initialRows, int32_t initialCols) :
+        _historySize(historySize),
+        _initialRows(initialRows),
+        _initialCols(initialCols)
+    { }
+
+    // property getters - all implemented
+    int32_t HistorySize() { return _historySize; }
+    int32_t InitialRows() { return _initialRows; }
+    int32_t InitialCols() { return _initialCols; }
+    uint32_t DefaultForeground() { return 0xFFFFFFFFUL; }
+    uint32_t DefaultBackground() { return 0UL; }
+    bool SnapOnInput() { return false; }
+    uint32_t CursorColor() { return 0xFFFFFFFFUL; }
+    CursorStyle CursorShape() const noexcept { return CursorStyle::Vintage; }
+    uint32_t CursorHeight() { return 42UL; }
+
+    // other implemented methods
+    uint32_t GetColorTableEntry(int32_t) const { return 123; }
+
+    // property setters - all unimplemented
+    void HistorySize(int32_t) { }
+    void InitialRows(int32_t) { }
+    void InitialCols(int32_t) { }
+    void DefaultForeground(uint32_t) { }
+    void DefaultBackground(uint32_t) { }
+    void SnapOnInput(bool) { }
+    void CursorColor(uint32_t) { }
+    void CursorShape(CursorStyle const&) noexcept { }
+    void CursorHeight(uint32_t) { }
+
+    // other unimplemented methods
+    void SetColorTableEntry(int32_t /* index */, uint32_t /* value */) { }
+
+private:
+    int32_t _historySize;
+    int32_t _initialRows;
+    int32_t _initialCols;
+};
+
+#define WCS(x) WCSHELPER(x)

```

Comment:
Any reason that this firs macro just expands into the second? Couldn't it just be `#define WCS(x) L#x`?

Reply:
woah well that's... insane, but I get it. Such is life using C macros :P